src/ansi-converter.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from "vscode";
+import AnsiColorsParser, { ParsedColor, ParsedSpan, parse } from "ansicolor";
+
+// taken from the ansicolor source code
+const colorNameAsBright = {
+  'red': 'lightRed',
+  'green': 'lightGreen',
+  'yellow': 'lightYellow',
+  'blue': 'lightBlue',
+  'magenta': 'lightMagenta',
+  'cyan': 'lightCyan',
+  'black': 'darkGray',
+  'lightGray': 'white'
+}
+
+export function doesTextContainAnsiCodes(text: string): boolean {
+  return AnsiColorsParser.isEscaped(text);
+}
+
+export function convertAnsiTextToHtml(text: string): string {
+  const parsed = parse(text).spans;
+
+  // According to VS Code source code, only span supports style
+  // https://github.com/microsoft/vscode/blob/6d2920473c6f13759c978dd89104c4270a83422d/src/vs/base/browser/markdownRenderer.ts#L301
+  const html = parsed.map((span) => `<span style="${getSupportedCSSFromAnsiSpan(span)}">${span.text.replaceAll('\n', '<br/>')}</span>`).join("");
+
+  return html;
+}
+
+function getSupportedCSSFromAnsiSpan(span: ParsedSpan): string {
+  // According to VS Code source code:
+  // 1. span only support color and background-color in style
+  // 2. colors must be in hex format
+  // 3. color must be first and than the background-color
+  // 4. there must be a semicolon at the end
+  // 5. color and background-color are optional
+  // 6. must be no spaces
+  //
+  // Source: https://github.com/microsoft/vscode/blob/6d2920473c6f13759c978dd89104c4270a83422d/src/vs/base/browser/markdownRenderer.ts#L309
+
+  const css: string[] = [];
+
+  if (span.color) {
+    const color = getColor(span.color, 'color', !!span.bgColor);
+
+    if (color) {
+      css.push(`color:${color};`);
+    }
+  }
+
+  if (span.bgColor) {
+    const backgroundColor = getColor(span.bgColor, 'background', true);
+
+    if (backgroundColor) {
+      css.push(`background-color:${backgroundColor}`);
+    }
+  }
+
+  return css.join("");
+}
+
+function getColor(color: ParsedColor, type: 'color' | 'background', hasBackground: boolean): string | undefined {
+  const isLightTheme = vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Light || vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.HighContrastLight;
+
+  if (!color.name) {
+    const shouldHaveHalfTransparentText = color.dim && (type === 'background' || (type === 'color' && !hasBackground));
+
+    // 50% transparent black / white depend on the theme
+    return shouldHaveHalfTransparentText ? isLightTheme ? '#00000080' : '#FFFFFF80' : undefined;
+  }
+
+  const colorName = color.bright ? colorNameAsBright[color.name as keyof typeof colorNameAsBright] : color.name;
+
+  const rgb = AnsiColorsParser.rgb[colorName as keyof typeof AnsiColorsParser.rgb];
+
+  const alphaPartInHex = color.dim ? '80' : 'FF';
+
+  let hexColor: string;
+
+  // Modify the common colors to be nicer
+  // TODO - should somehow support the theme colors
```

Comment:
[As of last week](https://github.com/microsoft/vscode/commit/126ded71ed6a2b74f142e3b6362ed2121046fb46) (currently only on VS Code Insiders) you can use the --vscode theme color variables here, including the [integrated terminal colors](https://code.visualstudio.com/api/references/theme-color#integrated-terminal-colors).

This would only cause issues if the terminal background color is configured to be a different type than the panel or peek backgrounds, but this is not common and using these variables is probably the best we can do

Reply:
I tried to make it work for hours when making the PR, because this is rendered as markdown and not HTML I can't use the variables

If we could have HTML output that would be great

Reply:
That's a permalink to a very old commit, the latest is https://github.com/microsoft/vscode/blob/97c8d7e7ccf9028bba26ea47d284bbce93e8a1e8/src/vs/base/browser/markdownRenderer.ts#L382